### PR TITLE
refactor: move server.id for MySQL CDC from meta to fe

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
motivation is to simplify create stream job process in meta (prepare for  #19171, where we will completely remove `fill_table_stream_graph_info`)

Also made a behavior change: previously, auto assign `server.id` will only for legacy `CREATE TABLE` CDC, but not shared `CREATE SOURCE`.

previous work: #13031

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by xxchan** [comment]
* **#19171** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/19171?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#19169** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/19169?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @xxchan and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/frontend/src/handler/create_source.rs [hunk 1]
```diff
@@ -1465,6 +1466,14 @@ pub fn bind_connector_props(
                 .to_string(),
         );
     }
+    if !is_create_source && with_properties.is_mysql_cdc_connector() {
```
Comment (StrikeW) [inline src/frontend/src/handler/create_source.rs:1469]: Should put in `if is_create_source && create_cdc_source_job {` branch, cdc Table doesn't need this option.

Comment (xxchan) [reply src/frontend/src/handler/create_source.rs:1469]: :eyes: https://github.com/risingwavelabs/risingwave/pull/13031 This PR is only for CDC table, but not (shared) CDC source.

Comment (xxchan) [reply src/frontend/src/handler/create_source.rs:1469]: Do you think it should apply to both `is_create_source` and `! is_create_source`?

File: src/frontend/src/handler/create_source.rs [hunk 2]
```diff
@@ -1465,6 +1466,14 @@ pub fn bind_connector_props(
...
+        // Generate a random server id for mysql cdc source if needed
+        // `server.id` (in the range from 1 to 2^32 - 1). This value MUST be unique across whole replication
```
Comment (BugenZhao) [inline src/frontend/src/handler/create_source.rs:1471]: How do we ensure uniqueness by generating a random number locally?

Comment (StrikeW) [reply src/frontend/src/handler/create_source.rs:1471]: We don't guarantee uniqueness, this is just a side note.

File: src/frontend/src/handler/create_source.rs (hunk 1 above)
Comment (StrikeW) [reply src/frontend/src/handler/create_source.rs:1469]: > 👀 #13031 This PR is only for CDC table, but not (shared) CDC source.

You are right. This for the old syntax previously, I think we can apply to both `is_create_source` and ! `is_create_source`.

File: e2e_test/source_inline/cdc/mysql/mysql_create_drop.slt.serial [hunk 3]
```diff
@@ -45,7 +45,6 @@ create source s with (
   username = 'shared-cdc',
   password = 'abcdef',
   database.name = 'testdb1',
-  server.id = '114514'
```
Comment (xxchan) [inline e2e_test/source_inline/cdc/mysql/mysql_create_drop.slt.serial:48]: Manually tested that before this PR, this test will fail if `server.id` not specified

**Review by StrikeW — APPROVED** [review]

File: src/frontend/src/handler/create_source.rs (hunk 2 above)
Comment (BugenZhao) [reply src/frontend/src/handler/create_source.rs:1471]: > this is just a side note

I don't quite get this. So will this be a problem?

**Pull request merged**